### PR TITLE
Feature/issue 2

### LINKED ISSUE #2: Save Image Link
#### As a user I want to save a link to an image.

_Assignees_: @HaominZhang, @XanderStrike

__Story__:

Now that we have an awesome landing page for our users to experience it is time
to introduce our first feature! In general, we want to deliver value to our
customers as fast as possible. As a result, we have decided that we want our
users to be able to save links to images, and immediately see that we have
saved the URL to the image.

One could argue that simply saving the image, and not showing it, provides some
value. Yes, you could split this story here. However, because saving image URLs
and viewing those images are both relatively small additions that work well
together, it is acceptable to complete them together. Note, on the other hand,
that we do not yet want to implement the image index. Its addition adds too
much complexity, so we’ll save that for a future story.

In general, feature addition, or stories, should be split into the smallest
deliverable chunk that provides value to your users with the caveat that it
might make sense to combine two, sometimes more, tiny and related features into
a single story.

__Acceptance criteria__:
- [x] The landing page links to the image link submission form.
- [x] The link is entered through a form.
- [x] After the form is saved, the link is persisted in the database.
- [x] After the form is submitted, the user is redirected to a page displaying
  the image.
- [x] I cannot successfully save an image with an invalid URL.
- [x] An error message is associated with the appropriate input field on
  failure.

__Discussion Topic__:
- [x] What is a valid image URL?

__Dependencies__:
- Deployment Environment

__Note__: Many image sharing sites will permit one to upload images, or
download images when provided a link. For this project we will only work with
the link, i.e., URL, to various images.

__Note__: To understand the database tables created by the `acts_as_taggable` gem, you may find it useful to use a program like the [DB Browser for SQLite](http://sqlitebrowser.org/).

### PR DESCRIPTION
**Pull request by chefnobody**
Closes #2

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -18,6 +18,7 @@ gem 'jquery-rails'
 # gem 'bcrypt', '~> 3.1.7'
 
 gem 'simple_form'
+gem "validate_url"
```
Comment (blakeherrington) [inline Gemfile]: Where is the change to actually add the url validation to the model and corresponding tests?

File: test/models/image_test.rb [hunk 2]
```diff
@@ -5,4 +5,9 @@ class ImageTest < ActiveSupport::TestCase
     img = Image.create(url: 'http://pizzahut.com')
     assert_equal 'http://pizzahut.com', img.url
   end
+
+  test 'Image creation fails with invalid url' do
+    image = Image.create(url: nil)
+    assert_equal ["can't be blank"], image.errors[:url]
```
[the remark this answers is not in the document]
Comment (chefnobody) [reply test/models/image_test.rb]: Gotcha. Makes sense. I like that approach.

File: Gemfile (hunk 1 above)
Comment (chefnobody) [reply Gemfile]: URL validation is here: https://github.com/AppFolioOnboarding/image-share-chefnobody/pull/14/commits/a6873e2d19c0b4f5282162bc7776d242bb3860df#diff-69e1773ce93d7ede3d3d0c7f4e6c7024R2

Image model test is here: https://github.com/AppFolioOnboarding/image-share-chefnobody/pull/14/commits/a6873e2d19c0b4f5282162bc7776d242bb3860df#diff-34a98d6e61262613f70be59b95885ab4R9

I think what happened was I committed and pushed the implementation of the validation (+ tests) before committing and pushing _this_ commit that pulls in the validation gem. My bad.

Comment (chefnobody) [reply Gemfile]: ... and looking back it would have been better to include the usage of the validator, plus the model tests in one commit.

**Pull request merged**